### PR TITLE
Covid19api

### PR DESCRIPTION
**Pull request by fsan110**
Github Issue #124 

## PR Type

What kind of change does this PR introduce?

-  Feature

## What is the current behavior?

- /covid API not implemented

## What is the new behavior?

- /covid API is implemented and is available for front-end use

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo`
- [x] Added new tests to the test suite for the feature
- [x] Tested the full test suite, and ensured that the feature does not break the current build.
- [ ] Docs have been added/updated which fit (for bug fixes/features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [x] Checked licensing of Frameworks/Libraries (if applicable)

    -  Apace Commons IO - Apache 2.0
    -  JSoup - MIT

### CODE REVIEW

**Comment by jonathanvenema98** [comment]
Has been updated to reflect changes on Master branch, has had a test suite implemented, and fixed issues with MultiFeedController to fix the failing check.

**Comment by fsan110** [comment]
> Hey, aside from what we discussed, I think we could separate the getFeed() function into smaller functions. for example, have separate functions for retrieving the data from the website, and then another function for parsing of the data from the first function, and then finally your getFeed() function which returns the flux object. this would make testing easier

File: server/src/test/java/com/feeder/server/controller/MultiFeedControllerIntegrationTest.java [hunk 1]
```diff
@@ -18,7 +23,7 @@
   @MockBean FeedProvider<HackerNewsData> mockHackerNewsFeedProvider;
   @MockBean FeedProvider<GithubData> mockGithubFeedProvider;
   @MockBean FeedProvider<TwitterData> mockTwitterFeedProvider;
-  @MockBean FeedProvider<WeatherData> mockWeatherFeedProvider;
```
Comment (lucycy98) [inline server/src/test/java/com/feeder/server/controller/MultiFeedControllerIntegrationTest.java:21]: hey any reason why you got rid of this?

File: server/src/main/java/com/feeder/server/controller/MultiFeedController.java [hunk 2]
```diff
@@ -53,13 +48,13 @@
     return twitterFeedProvider.getFeed();
   }
 
-  @GetMapping("/weather")
```
Comment (lucycy98) [inline server/src/main/java/com/feeder/server/controller/MultiFeedController.java:56]: can we add this line back?

File: server/src/main/java/com/feeder/server/controller/MultiFeedController.java [hunk 3]
```diff
@@ -26,11 +20,12 @@
   @Autowired private FeedProvider<GithubData> githubFeedProvider;
   @Autowired private FeedProvider<HackerNewsData> hackerNewsFeedProvider;
   @Autowired private FeedProvider<TwitterData> twitterFeedProvider;
-  @Autowired private FeedProvider<WeatherData> weatherDataProvider;
+  @Autowired private FeedProvider<CovidNineteenData> covidNineteenFeedProvider;
 
   @GetMapping("/")
   public Flux<? extends GenericData> multiFeedMixerFlow() {
-    return Flux.merge(redditFlow(), githubFlow(), hackerNewsFlow(), twitterFlow(), weatherFlow());
```
Comment (lucycy98) [inline server/src/main/java/com/feeder/server/controller/MultiFeedController.java:33]: and this

**Pull request merged**